util/err.c

```diff
@@ -65,3 +66,38 @@ void libutil_errx(int eval, const char *fmt, ...) {
 
     verrx(eval, fmt, args);
 }
+
+extern int getargs(void *, int, char *, int);
+
+static char buffer[PATH_MAX];
+static const char* __progname = NULL;
+
+const char* getprogname (void) {
+
+    if (!__progname) {
+        struct procentry64 entry;
+        pid_t pid = getpid();
+
+        if(getprocs64(&entry, sizeof(entry), NULL, 0, &pid, 1) > 0)
+        {
+            if(getargs(&entry, sizeof(entry), buffer, sizeof(buffer)) == 0)
+            {
+                const char *p;
+                __progname = buffer;
+                p = strrchr (__progname, '/');
+                if (p != NULL)
+                    __progname = p + 1;
+            }
+        }
+    }
+
```

Comment:
```suggestion
```

https://www.unix.com/man-page/FreeBSD/3/getprogname/

     The getprogname() function returns the name of the program.  If the name has not been set yet, it will return NULL.